File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        
        int n = nums.length;
        
        int majorityElement = -1;
        int count = 0;
        
        for(int num: nums){
            
            if(count == 0){
                majorityElement = num;
                count = 1;
            }
            else if(majorityElement == num){
                count++;
            }
            else{
                count--;
            }
        }
        
        count = 0;
        for(int num: nums){
            if(majorityElement == num){
                count++;
            }
        }
        
        if(count > n/2)
            return majorityElement;
        
        return -1;
    }
}

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        
        int n = nums.size();
        
        int el ;
        int cnt = 0;
        
        for(int i=0; i<n; i++){
            if(cnt == 0){
                cnt = 1;
                el = nums[i];
            }
            else if(el == nums[i]){
                cnt++;
            }
            else{
                cnt--;
            }
        }
        
        cnt = 0;
        for(int i=0; i<n; i++){
            if(el == nums[i])
                cnt++;
        }
        
        if(cnt > n/2)
            return el;
        
        return -1;
    }
};